tests/test_serving.py

```diff
@@ -436,3 +436,33 @@ def app(environ, start_response):
     assert res.read() == b'YES'
 
     conn.close()
+
+
+def test_multiple_headers_concatenated_per_rfc_3875_section_4_1_18(dev_server):
+    server = dev_server(r'''
+    from werkzeug.wrappers import Response
+    def app(environ, start_response):
+        start_response('200 OK', [('Content-Type', 'text/plain')])
+        return [environ['HTTP_XYZ'].encode()]
+    ''')
+
+    if sys.version_info[0] == 2:
+        from httplib import HTTPConnection
+    else:
+        from http.client import HTTPConnection
+
+    conn = HTTPConnection('127.0.0.1', server.port)
+    conn.connect()
+    conn.putrequest('GET', '/')
+    conn.putheader('Accept', 'text/plain')
+    conn.putheader('XYZ', 'a')
+    conn.putheader('XYZ', 'b')
+    conn.putheader('XYZ', 'c')
```

Comment:
Does the order of headers make any difference? E.g.:

    conn.putheader('XYZ', 'a')
    conn.putheader('SOMETHING_ELSE', 'foo')
    conn.putheader('XYZ', 'b')